enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,279 @@
+* Ensure no performance regression occurs when replacing Prometheus Adapter with Metrics Server
+* Improved scale up time for autoscaling
+
+### Non-Goals
+
+* Allow optional cluster monitoring stack
+* Moving `openshift-monitoring/cluster-monitoring-config` config-map to a Custom Resource
+
+## Proposal
+
+The proposal is to replace the Prometheus Adapter with the Metrics Server for monitoring resource metrics in OpenShift.
+
+### Workflow Description
+
+The workflow for using the Metrics Server in OpenShift for monitoring resource metrics would involve the following steps:
+
+1. User sets `enabled: true` in the ConfigMap `openshift-monitoring/cluster-monitoring-config` under `metricsServer` section
+
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: cluster-monitoring-config
+      namespace: openshift-monitoring
+    data:
+      config.yaml: |
+        metricsServer:
+          enabled: true
+    ```
+
+    With this configuration, the cluster monitoring operator should:
+    1. Deploy the `metrics-server` resources in the `openshift-monitoring` namespace
+    1. Verify that the `metrics-server` service is functional by querying the `/readyz` endpoint
+    1. Wait for `5 x metrics-server's scrape interval` to have enough data populated in `metrics-server` in-memory
+    1. Update `APIService` resource to point to `metrics-server` API instead of `prometheus-adapter` API
+    1. Delete `prometheus-adapter` resources
+
+1. User sets `enabled: false` in the ConfigMap `openshift-monitoring/cluster-monitoring-config` under `metricsServer` section
+
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: cluster-monitoring-config
+      namespace: openshift-monitoring
+    data:
+      config.yaml: |
+        metricsServer:
+          enabled: false
+    ```
+
+  With this configuration, the cluster monitoring operator should:
+  1. Deploy the `prometheus-adapter` resources in the `openshift-monitoring` namespace
+  1. Verify that the `prometheus-adapter` service is functional by querying the `/readyz` endpoint
+  1. Update `APIService` resource to point to `prometheus-adapter` API instead of `metrics-server` API
+  1. Delete `metrics-server` resources
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- We propose to add type `metricsServer` to the `openshift-montoring/cluster-monitoring-config` configmap to define 
+  metrics-server settings. 
+
+  ```go
+  type ClusterMonitoringConfiguration struct {
+    ...
+    // `MetricsServerConfig` defines settings for the MetricsServer component.
+    MetricsServerConfig *MetricsServerConfig `json:"metricsServer,omitempty"`
+  }
+  ```
+
+  When `enabled` is set `true`, the cluster monitoring operator configures the Metrics Server resources in `openshift-monitoring` namespace and tears down the Prometheus adapter resources once Metrics Server is fully functional.
```

Comment:
As said above by @slashpai there will be no `enabled`  setting for metrics-server. The migration should happen automatically for all customers when the feature will be GA. 
CMO should probably reuse the prometheus-adapter's settings such as nodeSelector, tolerations and resources when the switch happens to avoid unpleasant surprises for users (like they might expect that monitoring components are scheduled on infra nodes).